plugins/node/opentelemetry-instrumentation-dataloader/src/instrumentation.ts

```diff
@@ -0,0 +1,228 @@
+
+          const parent = context.active();
+          const span = self.tracer.startSpan(
+            `${MODULE_NAME}.batch`,
+            {
+              links: this._batch?.spanLinks as Link[] | undefined,
+            },
+            parent
+          );
+
+          return context.with(trace.setSpan(parent, span), () => {
+            return (original.apply(this, args) as Promise<unknown[]>)
+              .then(value => {
+                span.end();
+                return value;
+              })
+              .catch(err => {
+                span.recordException(err);
+                span.setStatus({
+                  code: SpanStatusCode.ERROR,
+                  message: err.message,
+                });
+                span.end();
+                throw err;
+              });
+          });
+        }
+      });
+
+      return inst;
+    }
+
+    PatchedDataloader.prototype = prototype;
+    return PatchedDataloader as unknown as typeof Dataloader;
+  }
+
+  private _patchLoad(proto: typeof Dataloader.prototype) {
+    if (isWrapped(proto.load)) {
+      this._unwrap(proto, 'load');
+    }
+
+    this._wrap(proto, 'load', this._getPatchedLoad.bind(this));
+  }
+
+  private _getPatchedLoad(original: LoadFn): LoadFn {
+    const instrumentation = this;
+
+    return function patchedLoad(
+      this: typeof Dataloader.prototype,
+      ...args: Parameters<typeof original>
+    ) {
+      const parent = context.active();
+      const span = instrumentation.tracer.startSpan(
+        `${MODULE_NAME}.load`,
+        { kind: SpanKind.CLIENT },
+        parent
+      );
+
+      return context.with(trace.setSpan(parent, span), () => {
+        const result = original
+          .call(this, ...args)
+          .then(value => {
+            span.end();
+            return value;
+          })
+          .catch(err => {
+            span.recordException(err);
+            span.setStatus({
+              code: SpanStatusCode.ERROR,
+              message: err.message,
+            });
+            span.end();
+            throw err;
+          });
+
+        const loader = this as DataloaderInternal;
+
+        if (loader._batch) {
+          if (!loader._batch.spanLinks) {
+            loader._batch.spanLinks = [];
+          }
+
+          loader._batch.spanLinks.push({ context: span.spanContext() } as Link);
+        }
+
+        return result;
+      });
+    };
+  }
+
+  private _patchLoadMany(proto: typeof Dataloader.prototype) {
+    if (isWrapped(proto.loadMany)) {
+      this._unwrap(proto, 'loadMany');
+    }
+
+    this._wrap(proto, 'loadMany', this._getPatchedLoadMany.bind(this));
+  }
+
+  private _getPatchedLoadMany(original: LoadManyFn): LoadManyFn {
+    const instrumentation = this;
+
+    return function patchedLoadMany(
+      this: typeof Dataloader.prototype,
+      ...args: Parameters<typeof original>
+    ) {
+      const parent = context.active();
+      const span = instrumentation.tracer.startSpan(
```

Comment:
In quite some instrumentations for client libraries a check is done if parent span is `undefined` and create new spans only if it isn't. Some have also a config option like `requireParent`.

There is no clear right/wrong regarding this and it can be added later also. So please take this just as hint to think about what you would expect here.

Reply:
Good idea, will add this option, will default to false (i.e always creating spans)